src/components/controllers/slotted-controller.ts

```diff
@@ -19,19 +23,40 @@ export class SlottedController implements ReactiveController {
   }
 
   onSlotChange() {
-    const elementName = this.hostEl.tagName;
-    const myElement: HTMLElement | null = this.hostEl.parentElement
-      ? this.hostEl.parentElement.querySelector(elementName)
-      : null;
-    if (myElement && myElement.shadowRoot) {
-      const slot: HTMLElement | null = myElement.shadowRoot.querySelector(
-        'slot'
-      )
-        ? myElement.shadowRoot.querySelector('slot')
-        : null;
-      if (slot) {
-        slot.append(...myElement.children);
-      }
+    // Search for all slots provided by the Web Component.
+    const slotsArray = this.hostEl.renderRoot.querySelectorAll('slot');
```

Comment:
Is there a better name for this or can you convert this to an Array?

https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll

This is actually a `NodeList` of `Element`.

Why not just `slots` or `slotElements`?

Reply:
Agreed on clarification of variable name.